subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -310,6 +313,25 @@ static int wifi_set_twt(uint32_t mgmt_request, struct net_if *iface,
 		goto fail;
 	}
 
+	if_cfg = net_if_config_get(iface);
+
+	if (if_cfg) {
+		for (i = 0; i < NET_IF_MAX_IPV4_ADDR; i++) {
+			if (if_cfg->ip.ipv4->unicast[i].addr_type ==
```

Comment:
What about static IP? We should only be enforcing having an IP not the type of IP.

Reply:
and what about IPv6 which is used by Matter and others?

Reply:
Yeah we can check IP.
But for IPv6, by default an address is assigned(AUTOCONF), So check will fail for IPv6.

Reply:
> But for IPv6, by default an address is assigned(AUTOCONF), So check will fail for IPv6.

Why would the check fail? I don't understand. If IPv6 auto-configuration is enabled then it will always have a link local IP, but still it has to wait for `RS` to get the global prefix else it cannot communicate with the network and auto-configuration itself can be disabled and we might end up using DHCPv6, so, many possibilities.

At the very least check for valid IPv4 or Global IPv6, this is kind of addressed by the `config` module (if enabled). See https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/config/init.c#L241.

Having this check in Wi-Fi stack is good, but its more of upper layers responsibility. So, once should we make this optional with a Kconfig? or we can enable by default and provide a Kconfig override that turns this in to a warning? Either way is fine with me.